torch/_dynamo/guards.py

```diff
@@ -453,8 +465,8 @@ def NN_MODULE_PARAM_NAMES(self, guard):
 
         code = list()
         code.append(f"___check_type_id({ref}, {self.id_ref(t)})")
-        code.append(f"{{k for k, v in {ref}.named_parameters()}} == {keys!r}")
-
+        if keys:
+            code.append(f"{{k for k, v in {ref}.named_parameters()}} == {keys!r}")
```

Comment:
this guard is effectively unused because of the reason above - we skip nn module guards unless a (very tired) config is set.

Add a test for the recompilation of when we a buffer goes from tensor -> int.

Reply:
Good point!